src/nvim/option.c

```diff
@@ -3822,8 +3865,25 @@ set_bool_option (
    * End of handling side effects for bool options.
    */
 
+  // after handling side effects, call autocommand
+
   options[opt_idx].flags |= P_WAS_SET;
 
+  if (!starting) {
+    char_u buf_old[2], buf_new[2], buf_type[7];
+    vim_snprintf((char *)buf_old, 2, "%d", old_value ? true: false);
+    vim_snprintf((char *)buf_new, 2, "%d", value ? true: false);
```

Comment:
Thus you will have troubles passing `ULL` literal to `%d` format specifier (not a subject to integer promotions which are part of default promotions because `ULL` does not fit in `unsigned`), but not `_Bool` there (_Bool fits in `int`, so it will be promoted to `int`).